var/spack/repos/builtin/packages/mercurial/package.py

```diff
@@ -47,7 +48,8 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    depends_on('python@2.7:2.8,3.5:3.5.999,3.6.2:', when='@4.4:')
```

Comment:
What about Mercurial 4.3?

Reply:
I really don't know. I can add a comment, but if someone adds that she would probably know that.